crates/sui-graphql-rpc/src/server/data_provider.rs

```diff
@@ -216,6 +221,108 @@ fn convert_bal(b: sui_json_rpc_types::Balance) -> Balance {
         total_balance: BigInt::from_str(&format!("{}", b.total_balance)).unwrap(),
     }
 }
+pub(crate) async fn fetch_gas_input(cl: &SuiClient, digest: &String) -> Result<GasInput> {
+    let tx_block = cl
+        .read_api()
+        .get_transaction_with_options(
+            TransactionDigest::from_str(digest)?,
+            SuiTransactionBlockResponseOptions::full_content(),
+        )
+        .await?;
+    let tx = tx_block.transaction.unwrap().data;
+    let gas_data = tx.gas_data();
+
+    let payment_obj_ids = gas_data
+        .payment
+        .clone()
+        .iter()
+        .map(|x| x.object_id)
+        .collect::<Vec<_>>();
+
+    let obj_responses: Vec<SuiObjectResponse> =
+        future::try_join_all(payment_obj_ids.iter().map(|x| {
+            cl.read_api()
+                .get_object_with_options(*x, SuiObjectDataOptions::full_content())
+        }))
+        .await
+        .unwrap();
+
+    let payment_objs = obj_responses
+        .iter()
+        .map(|x| convert_obj(&x.data.as_ref().unwrap()))
+        .collect::<Vec<Object>>();
+
+    Ok(GasInput {
+        gas_sponsor: Some(Address::from(SuiAddress::from(gas_data.owner))),
+        gas_payment: Some(payment_objs),
+
+        gas_price: Some(
+            <BigInt as ScalarType>::parse(Value::String(gas_data.price.to_string())).unwrap(),
```

Comment:
It's probably worth adding a `parse_scalar` helper to `BigInt` to avoid introducing this cast syntax everywhere (similar for `<BigInt as ValueType>::parse` and `parse_value` if we find ourselves calling that often).

Reply:
Alternatively/better -- implement `From<u64>` etc for `BigInt`.